alchemy-web/src/content/docs/providers/neon/index.md

```diff
@@ -0,0 +1,314 @@
+
+### Using Encrypted Secrets
+
+For better security, use Alchemy's secret encryption:
+
+```ts
+import { alchemy } from "alchemy";
+import { NeonProject } from "alchemy/neon";
+
+const app = await alchemy("my-app");
+
+const project = await NeonProject("db", {
+  name: "My Database",
+  apiKey: alchemy.secret.env.NEON_API_KEY, // Encrypted in state
+});
+
+await app.finalize();
+```
+
+### Overriding Per Resource
+
+You can override authentication for individual resources to support multiple Neon accounts:
+
+```ts
+const project1 = await NeonProject("db1", {
+  name: "Project 1",
+  apiKey: alchemy.secret.env.NEON_API_KEY_ACCOUNT_1,
+});
+
+const project2 = await NeonProject("db2", {
+  name: "Project 2",
+  apiKey: alchemy.secret.env.NEON_API_KEY_ACCOUNT_2,
+});
+```
+
+## Example Usage
+
+### Basic Project and Branch
+
+```ts
+import { alchemy } from "alchemy";
+import { NeonProject, NeonBranch } from "alchemy/neon";
+
+const app = await alchemy("my-app");
+
+// Create a Neon project
+const project = await NeonProject("db", {
+  name: "My Database",
+  region_id: "aws-us-east-1",
+  pg_version: 16,
+});
+
+// Create a development branch
+const dev = await NeonBranch("dev", {
+  project,
+  name: "development",
+  endpoints: [
+    { type: "read_write" }
+  ],
+});
+
+console.log("Connection URI:", dev.connectionUris[0].connection_uri);
+
+await app.finalize();
+```
+
+### Git-Style Database Workflow
+
+```ts
+import { alchemy } from "alchemy";
+import { NeonProject, NeonBranch } from "alchemy/neon";
+
+const app = await alchemy("my-app");
+
+// Create project
+const project = await NeonProject("db", {
+  name: "My App Database",
+});
+
+// Main production branch (created automatically with project)
+const mainBranchId = project.branch.id;
+
+// Development branch from main
+const dev = await NeonBranch("dev", {
+  project,
+  name: "development",
+  parentBranch: mainBranchId,
+  endpoints: [
+    { type: "read_write" }
+  ],
+});
+
+// Feature branch from dev
+const feature = await NeonBranch("feature", {
+  project,
+  name: "feature-new-auth",
+  parentBranch: dev,
+  endpoints: [
+    { type: "read_write" }
+  ],
+});
+
+console.log("Feature branch connection:", feature.connectionUris[0].connection_uri);
+
+await app.finalize();
+```
+
+### Preview Deployments
+
+```ts
+import { alchemy } from "alchemy";
+import { NeonProject, NeonBranch } from "alchemy/neon";
+
+const app = await alchemy("my-app");
+
+const project = await NeonProject("db", {
+  name: "My App Database",
+});
+
+// Get PR number from environment (e.g., GitHub Actions)
+const prNumber = process.env.PR_NUMBER;
+
+// Create temporary preview branch that expires in 7 days
+const expiresAt = new Date(Date.now() + 7 * 24 * 60 * 60 * 1000);
+
+const preview = await NeonBranch(`preview-${prNumber}`, {
+  project,
+  name: `preview-pr-${prNumber}`,
+  parentBranch: project.branch.id,
+  expiresAt: expiresAt.toISOString(),
+  endpoints: [
+    { type: "read_write" }
+  ],
+});
+
```

Comment:
why does it need to expire, if this runs in a PR environment it should be torn down once the PR is merged. If a pr is open for more than 7 days this could cause problems no?

Reply:
I guess with alchemy there isn't really a use case for expiresAt (maybe there is?) but the api supports it, your call if it should be included or not.